vm/src/obj/objtype.rs

```diff
@@ -327,6 +334,17 @@ impl PyClassRef {
             }
         }
 
+        if let Some(initter) = typ.get_super_attr("__init_subclass__") {
+            let initter = PyClassMethod::descr_get(
+                vm,
+                initter,
+                None,
+                OptionalArg::Present(typ.clone().into_object()),
+            )?;
```

Comment:
Maybe use `vm.call_descr_get_specific()` here? `initter` won't necessarily always be a PyClassMethod, even though that's not correct, but we should probably still call the appropriate `__get__` descriptor.